### PR TITLE
added macOS compatibility to portmidi-3.17

### PR DESCRIPTION
**Pull request by plumps**
###### Motivation for this change
Needed portmidi for pygame. Haven't changed a thing about the missing java builds. 

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/portmidi/default.nix [hunk 1]
```diff
@@ -23,11 +25,11 @@ stdenv.mkDerivation rec {
   ];
 
   # XXX: This is to deactivate Java support.
-  patches = stdenv.lib.singleton (fetchurl rec {
+  patches = singleton (fetchurl rec {
     url = "https://raw.github.com/Rogentos/argent-gentoo/master/media-libs/"
         + "portmidi/files/portmidi-217-cmake-libdir-java-opts.patch";
     sha256 = "1jbjwan61iqq9fqfpq2a4fd30k3clg7a6j0gfgsw87r8c76kqf6h";
-  });
+  }) ++ optional stdenv.isDarwin ./darwin.patch;
```
Comment (LnL7) [inline pkgs/development/libraries/portmidi/default.nix]: Looks like you forgot to commit this

```
error: getting status of ‘/nix/store/hljk2wdrny0rzdx9krl1zrqh2disv0ha-git-export/pkgs/development/libraries/portmidi/darwin.patch’: No such file or directory
```

**Review by LnL7 — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/portmidi/default.nix (hunk 1 above)
Comment (plumps) [inline pkgs/development/libraries/portmidi/default.nix]: you are so right o.O., fixed that

File: pkgs/development/libraries/portmidi/default.nix [hunk 2]
```diff
@@ -44,14 +46,38 @@ stdenv.mkDerivation rec {
...
+  (path:
+  ''
+    substituteInPlace ./${path}/CMakeLists.txt --replace "CoreAudio.framework" \
+      "${CoreAudio}/Library/Frameworks/CoreAudio.framework"
+    substituteInPlace ./${path}/CMakeLists.txt --replace "CoreFoundation.framework" \
+      "${CF}/Library/Frameworks/CoreFoundation.framework"
+    substituteInPlace ./${path}/CMakeLists.txt --replace "CoreMIDI.framework" \
+      "${CoreMIDI}/Library/Frameworks/CoreMIDI.framework"
+    substituteInPlace ./${path}/CMakeLists.txt --replace "CoreServices.framework" \
+      "${CoreServices}/Library/Frameworks/CoreServices.framework"
```
Comment (LnL7) [inline pkgs/development/libraries/portmidi/default.nix]: This seems to replace the same things as the patch, why not replace it directly here?

Comment (LnL7) [inline pkgs/development/libraries/portmidi/default.nix]: To clarify, I think something like this should make most of the patch file redundant.

```
substituteInPlace pm_dylib/CMakeLists.txt \
    --replace '${FRAMEWORK_PATH}/CoreAudio.framework' "${CoreAudio}/Library/Frameworks/CoreAudio.framework"
...
```

**Comment by c0bw3b** [comment]
@plumps are you still working on this?

**Comment by plumps** [comment]
@c0bw3b nope

**Comment by c0bw3b** [comment]
@NixOS/darwin-maintainers does someone feel like resuming/completing this or should we close for now?

**Review by matthewbauer — CHANGES_REQUESTED** [review]
See @lnl7's suggestions

**Comment by c0bw3b** [comment]
Stalled

**Pull request closed without merging**